cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -531,7 +531,21 @@ SwiftGenerator::validateMetadata(const UnitPtr& u)
         .validOn = {typeid(Module)},
         // Even though it's really 'module:prefix' the validator sees this as a single argument since there's no commas.
         .acceptedArgumentKind = MetadataArgumentKind::SingleArgument,
-    };
+        .extraValidation = [](const MetadataPtr& metadata, const SyntaxTreeBasePtr& p) -> optional<string>
+        {
+            const string msg = "'swift:module' is deprecated; use 'swift:identifier' to remap modules instead";
+            p->unit()->warning(metadata->file(), metadata->line(), Deprecated, msg);
```

Comment:
We always emit a deprecation warning, but still return `nullopt` to signal that this metadata is fine...